aten/src/ATen/native/ReplicationPadding.cpp

```diff
@@ -761,9 +706,15 @@ Tensor& replication_pad3d_backward_out_cpu_template(
   int64_t oheight = iheight + ptop + pbottom;
   int64_t owidth  = iwidth + pleft + pright;
 
-
-  shapeCheck3d(input, pleft, pright,
```

Comment:
Do you need to add a call to `at::native::padding::check_valid_input` here too?

Reply:
I don't think so.

The original code does check with `shapeCheck3d` on both forward and backward.

But `shapeCheck3d` is checking the `input`, I don't think we need to re-check the `input` from a backward calling. The backward is not intended to be used directly by users, it is launched by autograd, we should have already checked the `input` in the forward. So i think the original check of input from the backward is a duplicate.

Reply:
@albanD - what do you think?

Reply:
fair enough, the input check is added back.